content/cryptography/basic-signatures/slides.md

```diff
@@ -12,10 +12,12 @@ duration: 1 hour
 
 Signature libraries should generally all expose some basic functions:
 
-- `fn generate_key(r) -> sk;` <br /> Generate a `sk` (secret key) from some input `r`.
-- `fn public_key(sk) -> pk;` <br /> Return the `pk` (public key) from a `sk`.
-- `fn sign(sk, msg) -> signature;` <br /> Takes `sk` and a message; returns a digital signature.
-- `fn verify(pk, msg, signature) -> bool;` <br /> For the inputs `pk`, a message, and a signature; returns whether the signature is valid.
+<ul>
```

Comment:
All this stuff doesn't even render for me.

What's the reasoning for making this as inline HTML instead of just keeping it as markdown? I don't see anything in here that isn't supported in markdown (unordered lists, code).

Reply:
It renders for me. But it already got pushed in the pba-content repo. Idk half of the slides in all modules is HTML or MD so it is a mix. There really hasnt been any enforcement. Also the MD cant be limiting sometimes it doesnt cover everything. If its really a big deal it can be changed. It wouldnt take long.

Reply:
Ah, weird that it doesn't render for me locally but does otherwise

Reply:
Glad to see the book getting some love :stars: wishing the team for HK all the best :pray: - hope to see the work for that wave end up translating here. I am not in the game for it (at least for now) but have some advice/notes if you wanna hear em :grin: 

> Idk half of the slides in all modules is HTML or MD so it is a mix.

Ultimately not up to me anymore, though I tried to enforce MD. I did work with you @coax1d directly on this before in case you forgot :stuck_out_tongue_closed_eyes: 🫶 - you even helped convert a subset of your HTML to MD where possible before too. I did intentionally _archive_ the (private) `pba-content` repo that is not AFAIK up to date with the fair bit of polish I put in post UCB that is now in the initial publishing of this book that included moving HTML to MD where possible. I highly suggest pulling content from this repo before making changes to the what IMHO should remain archived older repo.

For some background, the chief motivator for using the `reveal-md` tooling for us from the beginning was:

> **Focus on content**. Don't mess with style. You do not need to know anything but simple MD to make slides. Where MD is insufficiently expressive, there are minimal custom HTML slide style elements to get that content crafted and delivered, with easy templates/examples in https://github.com/Polkadot-Blockchain-Academy/pba-book/tree/main/content/contribute/copy-paste-slides.
>
> **Maintainability and uniformity**. The book has a large number of contributors with zero up to expert HTML/CSS experience, and the next faculty to take on teaching may be anywhere in that range. It's much more sustainable to have as simple and clean content for everyone to reasonable in the long term, and much more portable (like in the pure reveal-md tooling -> to this book).

Maybe @Asamartino wants to do something different now, and that driving factor isn't as important now. :shrug: Happy to help outline and implement a strategy with maintainers here that lands in the contributor's guide, if the PBA team wants this. 

---

For the rendering, sadly as it stands there are at least four different HTML servers depending on what commands you run (reveal has dev and build, bun with a minimal static server, and mdbook has a watching one). This could be cleaned up for a better overall UX, IMHO this would fork or replace the reveal-md tooling with something minimal here, and as I moved to bun for the book's static server here, I would opt for better configure mdbook (perhaps a custom plugin) as a tool to get slides -> reveal HTML -> full book. 

Maybe related: https://github.com/webpro/reveal-md/issues/433 but it's been a while since I played with different behavior.

In the end, the **source of truth on what things will render as should be assessed via running `markers` or `makers serve`** -> https://github.com/Polkadot-Blockchain-Academy/pba-book/blob/main/Makefile.toml#L108-L125 as this is what actually gets deployed to github pages.